rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java

```diff
@@ -413,13 +420,25 @@ public static boolean isFreeBridgeTx(Transaction rskTx, Constants constants, Act
      */
     public static boolean isContractTx(Transaction rskTx) {
         // Calls between contracts are done through internal transactions
-        return rskTx.getClass() == org.ethereum.vm.program.InternalTransaction.class;
+        return rskTx.getClass() == InternalTransaction.class;
     }
 
-    public static boolean isFromFederateMember(org.ethereum.core.Transaction rskTx, Federation federation, SignatureCache signatureCache) {
+    public static boolean isFromFederateMember(Transaction rskTx, Federation federation, SignatureCache signatureCache) {
         return federation.hasMemberWithRskAddress(rskTx.getSender(signatureCache).getBytes());
     }
 
+    /**
+     * Method that verify if an RskAddress is part of the Genesis Federation
+     *
+     * @param rskAddress
+     * @return boolean
+     */
+    private static boolean isFromGenesisFederation(RskAddress rskAddress) {
+        final List<BtcECKey> genesisFederationPublicKeys = bridgeConstants.getGenesisFederationPublicKeys();
+        final List<ECKey> genesisFederationECKeys = genesisFederationPublicKeys.stream().map(k -> ECKey.fromPublicOnly(k.getPubKey())).collect(Collectors.toList());
+        return genesisFederationECKeys.stream().anyMatch(m -> Arrays.equals(m.getAddress(), rskAddress.getBytes()));
```

Comment:
Perhaps this could be done in a single iteration. Convert from BtcEcKey to EcKey and then compare

Reply:
@marcos-iov Do you mean reduce these two lines in something like this?:

`genesisFederationPublicKeys.stream().anyMatch(k -> Arrays.equals((ECKey.fromPublicOnly(k.getPubKey())).getAddress(),rskAddress.getBytes()));`

Reply:
Yeah, but you can probably make it a bit more readable. Kind of 
```
genesisFederationPublicKeys.stream().anyMatch(genesisBtcPublicKey -> {
                ECKey genesisEcKey = ECKey.fromPublicOnly(genesisBtcPublicKey.getPubKey());
                return Arrays.equals(genesisEcKey.getAddress(), rskAddress.getBytes())
            }
        );
```